dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -86,15 +91,38 @@ void checkPermissions(MultiplePermissionsListener listener, Collection<String> p
     multiplePermissionsReport.clear();
     this.listener = listener;
 
-    startTransparentActivity();
+    startTransparentActivityIfNeeded();
   }
 
   /**
-   * Method called whenever the inner activity has been created and is ready to be used
+   * Check if there is a permission pending to be confirmed by the user and restarts the
+   * request for permission process.
    */
-  void onActivityCreated(Activity activity) {
-    this.activity = activity;
+  void checkPendingPermission(PermissionListener listener) {
+    MultiplePermissionsListenerToPermissionListenerAdapter adapter =
+        new MultiplePermissionsListenerToPermissionListenerAdapter(listener);
+    checkPendingPermissions(adapter);
+  }
+
+  /**
+   * Check if there are some permissions pending to be confirmed by the user and restarts the
+   * request for permission process.
+   */
+  void checkPendingPermissions(MultiplePermissionsListener listener) {
+    boolean shouldContinueRequestingPendingPermissions =
+        !pendingPermissions.isEmpty() && !rationaleAccepted.get();
+    this.listener = listener;
```

Comment:
Probably not, we are keeping a ref to something we are not going to use (and clean) so I'm moving it to the if block. Thanks!